playwright/_impl/_wait_helper.py

```diff
@@ -24,14 +24,43 @@
 
 
 class WaitHelper:
-    def __init__(self, channel_owner: ChannelOwner, api_name: str) -> None:
+    def __init__(self, channel_owner: ChannelOwner, event: str) -> None:
         self._result: asyncio.Future = asyncio.Future()
         self._wait_id = uuid.uuid4().hex
         self._loop = channel_owner._loop
         self._pending_tasks: List[Task] = []
-        self._channel_owner = channel_owner
+        self._channel = channel_owner._channel
         self._registered_listeners: List[Tuple[EventEmitter, str, Callable]] = []
-        channel_owner._wait_for_event_info_before(self._wait_id, api_name)
+        self._wait_for_event_info_before(self._wait_id, event)
+
+    def _wait_for_event_info_before(self, wait_id: str, event: str) -> None:
+        self._channel.send_no_reply(
+            "waitForEventInfo",
+            {
+                "info": {
+                    "waitId": wait_id,
+                    "phase": "before",
+                    "event": event,
+                }
+            },
+        )
+
+    async def _wait_for_event_info_after(
+        self, wait_id: str, error: Exception = None
+    ) -> None:
+        try:
+            await self._channel.send(
```

Comment:
You are changing the order of things with posting this in a task. Is there a reason to not send_no_reply?